src/Spago/Build.hs

```diff
@@ -200,33 +202,117 @@ run
 run maybeModuleName buildOpts extraArgs = do
   Config.Config { alternateBackend } <- view (the @Config)
   let moduleName = fromMaybe (ModuleName "Main") maybeModuleName
-  runBackend alternateBackend moduleName Nothing "Running failed; " buildOpts extraArgs
+  runBackend alternateBackend Nothing moduleName Nothing "Running failed; " buildOpts extraArgs
 
 
+-- | Run the select module as a script: init, compile, and run the provided module
+script
+  :: (HasEnv env)
+  => Text
+  -> Maybe Text
+  -> [PackageName]
+  -> RIO env ()
+script modulePath tag packageDeps = do
+  logDebug "Running `spago script`"
+  absoluteModulePath <- fmap Text.pack (makeAbsolute (Text.unpack modulePath))
+  moduleContents <- readTextFile (pathFromText absoluteModulePath)
+
+  moduleName <- case parseModuleName moduleContents of
+    Nothing -> die [ "Failed to extract module name from source file" ]
+    Just modName -> pure modName
+
+  GlobalCache cacheDir _ <- view (the @GlobalCache)
+  currentDir <- Turtle.pwd
+  Temp.withTempDirectory cacheDir "spago-script-tmp" $ \dir -> do
+    let tmpDir = Text.pack dir
+    Turtle.cd (Turtle.fromText tmpDir)
+
+    config@Config { packageSet = PackageSet{..}, ..}
+      <- Packages.initProject NoForce Dhall.WithComments tag
+
+    let newConfig :: Config
+        newConfig = config
+          { Config.dependencies = dependencies <> packageDeps
```

Comment:
By default all projects include `psci-support`, `effect`, and `console`. I think it merits discussion what packages are sensible to include by default for the `script` command. The last two seem reasonable, and there's good sense in keeping consistent, but `psci-support` seems obviously unnecessary.

Reply:
As part of the changes to construct the config in memory, I went ahead and changed it to include `effect`, `console`, and anything else the user specifies as a dependency.

Does this seem reasonable to y'all?

Reply:
Sounds great! I'd also add `prelude` in there by default